### ISSUE TITLE
Dependency conflict with Detox

### ISSUE THREAD

**Issue by simonmasinde** [mixpanel/mixpanel-react-native#173]
Error:
```
Dependency 'androidx.appcompat:appcompat:1.4.1' requires libraries and applications that
           depend on it to compile against version 31 or later of the
           Android APIs.

           :mixpanel-react-native is currently compiled against android-30.

**Comment by UpQuark** [comment]
Adding the following to `/android/build/gradle` to locally override the mixpanel target SDK is working for me with detox.
```
subprojects {
            if (subproject.name == 'mixpanel-react-native') {
                def String taskRequests = getGradle().getStartParameter().getTaskRequests().toString()
                if (taskRequests.contains("assembleAndroidTest")) {
                    android {
                        compileSdkVersion 33

                        defaultConfig {
                            targetSdkVersion 33
                        }
                    }
                }
            }
        }
    }
```

**Comment by dk-rp** [comment]
> Adding the following to `/android/build/gradle` to locally override the mixpanel target SDK is working for me with detox.
> 
> ```
> subprojects {
>             if (subproject.name == 'mixpanel-react-native') {
>                 def String taskRequests = getGradle().getStartParameter().getTaskRequests().toString()
>                 if (taskRequests.contains("assembleAndroidTest")) {
>                     android {
>                         compileSdkVersion 33
> 
>                         defaultConfig {
>                             targetSdkVersion 33
>                         }
>                     }
>                 }
>             }
>         }
>     }
> ```

@UpQuark bro I have added this in my android build.gradle but it is giving another error: 
Could not get unknown property 'subproject' for project ':app' of type org.gradle.api.Project. Can you help

**Comment by UpQuark** [comment]
Hey @dk-rp - I made corrective edits to my above comment that restore a missing line that actually gets you `subproject` (which I somehow deleted when pasting that snippet into my original comment), and indicates that the snippet is nested under `buildscripts` with my gradle plugin version. Hope that helps.

**Comment by zihejia** [comment]
Fixed in[ v2.3.0](https://github.com/mixpanel/mixpanel-react-native/releases/tag/v2.3.0). Thank you so much guys!
